browser-sign-in/OktaBrowserSignIn/WelcomeViewController.swift

```diff
@@ -15,70 +15,88 @@
  */
 
 import UIKit
-import OktaOidc
+import WebAuthenticationUI
 
 final class WelcomeViewController: UIViewController {
-
-    var oktaOidc: OktaOidc?
-    var stateManager: OktaOidcStateManager?
+    var auth: WebAuthentication?
     
+    @IBOutlet weak var signInButton: UIButton!
     @IBOutlet weak var ephemeralSwitch: UISwitch!
+    @IBOutlet weak var clientIdLabel: UILabel!
     
     override func viewDidLoad() {
         super.viewDidLoad()
         navigationItem.hidesBackButton = true
-        
-        if #available(iOS 13.0, *) {
-            // We'll allow the switch to be shown
+        if let configForUITests = configForUITests {
+            self.auth = WebAuthentication(
+                issuer: URL(string: configForUITests["issuer"]!)!,
+                clientId: configForUITests["clientId"]!,
+                scopes: configForUITests["scopes"]!,
+                redirectUri: URL(string: configForUITests["redirectUri"]!)!,
+                logoutRedirectUri: URL(string: configForUITests["logoutRedirectUri"] ?? ""),
+                additionalParameters: nil)
         } else {
-            ephemeralSwitch.superview?.isHidden = true
+            self.auth = WebAuthentication.shared
         }
         
-        do {
-            if let configForUITests = self.configForUITests {
-                oktaOidc = try OktaOidc(configuration: OktaOidcConfig(with: configForUITests))
-            } else {
-                oktaOidc = try OktaOidc()
-            }
-        } catch let error {
-            DispatchQueue.main.async {
-                let alert = UIAlertController(title: "Error", message: error.localizedDescription, preferredStyle: .alert)
-                alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
-                self.present(alert, animated: true, completion: nil)
-            }
-            return
+        if Credential.default != nil {
+            self.performSegue(withIdentifier: "show-details", sender: self)
```

Comment:
Does this mean at app launch, if you're signed in, you'll first see the welcome screen, followed by an animation to the details?